packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -1551,7 +1551,9 @@ export function cloneChildFibers(
   let currentChild = workInProgress.child;
   let newChild = createWorkInProgress(
     currentChild,
-    currentChild.pendingProps,
+    currentChild.pendingProps !== null
```

Comment:
So when is pendingProps null?

Reply:
We set it back to `null` in the complete phase, unless we tear (down-prioritize), in which case the torn props are stored on the current fiber. https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberCompleteWork.js#L395-L405

Reply:
@sebmarkbage Thanks for pointing that out; realized it's simpler to just never reset pending props.